sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/ObjectReplicationPolicy.java

```diff
@@ -31,10 +31,20 @@ public String getPolicyId() {
     }
 
     /**
-     * @return A {@code Map} of rules associated with this policy to the status of the replication associated with that
+     * @return A {@code List} of rules associated with this policy to the status of the replication associated with that
      * rule.
      */
-    public Map<String, String> getRules() {
-        return this.ruleStatuses;
+    public List<ObjectReplicationRule> getRules() {
+        return this.objectReplicationRules;
+    }
+
+    static int getIndexOfObjectReplicationPolicy(String policyId,
+        List<ObjectReplicationPolicy> objectReplicationPolicies) {
+        for (int i = 0; i < objectReplicationPolicies.size(); i++) {
+            if (policyId.equals(objectReplicationPolicies.get(i).getPolicyId())) {
+                return i;
+            }
+        }
+        return -1;
     }
```

Comment:
By looking at usage I think we could simplify this and code that's using it by usage of stream api.
https://stackoverflow.com/questions/23696317/find-first-element-by-predicate